src/main/java/com/powsybl/openloadflow/ac/ReactiveLimitsOuterLoop.java

```diff
@@ -63,35 +70,83 @@ private void switchPvPq(LfBus bus, EquationSystem equationSystem, VariableSet va
         }
     }
 
+    private enum ReactiveLimitDirection {
+        MIN,
+        MAX
+    }
+
+    private static final class PvToPqBus {
+
+        private final LfBus bus;
+
+        private final double q;
+
+        private final double qLimit;
+
+        private final ReactiveLimitDirection limitDirection;
+
+        private PvToPqBus(LfBus bus, double q, double qLimit, ReactiveLimitDirection limitDirection) {
+            this.bus = bus;
+            this.q = q;
+            this.qLimit = qLimit;
+            this.limitDirection = limitDirection;
+        }
+    }
+
+    private void switchPvPq(List<PvToPqBus> pvToPqBuses, EquationSystem equationSystem, VariableSet variableSet, int remainingPvBusCount) {
+        int modifiedRemainingPvBusCount = remainingPvBusCount;
+        if (modifiedRemainingPvBusCount == 0) {
+            // keep one bus PV, the strongest one which is one at the highest nominal level and highest active power
+            // target
+            PvToPqBus strongestPvToPqBus = pvToPqBuses.stream()
+                    .min(BY_NOMINAL_V_COMPARATOR
+                            .thenComparing(BY_TARGET_P_COMPARATOR)
```

Comment:
I am not sure to understand all here: you first sort by nominal voltage the candidate PV buses (in fact it is a way to select the buses that regulate the highest voltage) and then by highest active power. What happens if we just have one PV bus at 400 kV but not with the highest active power ?

Reply:
It is first sort on nominal V then target P, so yes if a generator is at higher nominalV but with a 1 MW targetP it will be chosen. Maybe I should remove the nominalV criteria and just focus on targetP?

Reply:
No I think it is better to rely on nominalV criteria. It is the nominal voltage of PV bus, so it means that it is the target V of the generator if remote ?

Reply:
No, in case of remove control, it is the nominal voltage of the generator not of the remote bus. I should probably fix this?

Reply:
Yes, I think it will be better.

Reply:
Done

Reply:
I have performed some tests and I will approve this PR. Just a question before: we always remove the strongest PV bus from switching or it is just when all generators of the network are about to switch ?